### DISCUSSION TITLE
force_indicate on Bluez backend

### DISCUSSION THREAD

**Discussion by h0ryz0n** [hbldh/bleak#1168] [Q&A]
* bleak version: 0.19.5
* Python version: 3.10.6
* Operating System: Ubuntu 22.04.1 LTS
* BlueZ version: 5.64

### Description

in the bleak.backends.winrt.client, the function start_notify has a force_indicate (bool) option, is there a way to acquire the same effect on the backend which uses BlueZ on Linux?

**Comment by dlech** [comment]
No, this is not possible. There is no parameter to [StartNotify](https://github.com/bluez/bluez/blob/35947e26771c89d965aec5ad059267bb8c15298e/doc/gatt-api.txt#L185). The code where BlueZ prefers notifications over indications is [here](https://github.com/bluez/bluez/blob/35947e26771c89d965aec5ad059267bb8c15298e/src/shared/gatt-client.c#L1583).

If you control the firmware of the peripheral, I would suggest not implementing notify and indicate on the same characteristic.

**Reply by h0ryz0n** [reply]
in characteristic.properties, the characteristic shows only 'indicate', can it be possible that the client.start_notify is still forcing the notifications? it's a bluez limitation?
thank you a lot for the support

**Reply by dlech** [reply]
If the characteristic only has the property flag set for indications and not notifications, the BlueZ should read that and enable indications.
